sky/backends/wheel_utils.py

```diff
@@ -129,7 +129,8 @@ def _build_sky_wheel() -> pathlib.Path:
 
         wheel_dir = WHEEL_DIR / hash_of_latest_wheel
         wheel_dir.mkdir(parents=True, exist_ok=True)
-        shutil.move(str(wheel_path), wheel_dir)
+        if not os.path.exists(os.path.join(wheel_dir, os.path.basename(wheel_path))):
+            shutil.move(str(wheel_path), wheel_dir)
```

Comment:
This seems indicating that the temp folder we are using in the code is not available for writing wheels to. Can we debug a bit for why the `/tmp` folder is not working and whether it will work with `tempfile.gettempdir()` instead of using the hardcoded `/tmp`.

Reply:
When I revert to the original code, the error I'm getting is: `shutil.Error: Destination path '/root/.sky/wheels/0b62cd7da6552f20e52d8a83687dd835/skypilot-1.0.0.dev0-py3-none-any.whl' already exists`. So I think it's actually that the wheel is persisting, and `shutil.move` fails to overwrite if it exists. Is it atypical that the wheel would already be found at the destination? 

I think there's a number of ways to fix this that are relatively harmless and idempotent. This could be checking if the file already exists and skipping (as I did here), or running a `shutil` method that overwrites instead. Just let me know what you're preference is.

Reply:
Hmm, this is weird. The original code's destination, `wheel_dir`, is a directory (supposedly to be `/root/.sky/wheels/0b62cd7da6552f20e52d8a83687dd835`) but the error message you shown `shutil.Error: Destination path '/root/.sky/wheels/0b62cd7da6552f20e52d8a83687dd835/skypilot-1.0.0.dev0-py3-none-any.whl' already exists` indicates the `wheel_dir` somehow becomes the target file.

IIUC, if the destination is a dir, `shutil` should correctly overwrite the file in it. Could you help check if that is the case in your case?

Reply:
Here's an example printout of the wheel_path, and wheel_dir:
`Wheel path: /tmp/tmp1ca8zhil/skypilot-1.0.0.dev0-py3-none-any.whl`
`Wheel dir: /root/.sky/wheels/6b3f8d33bac9d114a388c31be0e1998e`

I think it's possible that the 9p filesystem is again doing something odd here. Also from observation, it starts to fail the second time (not first time) the wheel hash changes. I probably need to do more digging here myself but let me know if that's helpful.

Reply:
So the reason this is failing seems simple. It turns out that `shutil.move` fails during a move across two filesystems. Since I'm moving the wheel from Modal's `tmp` mount to my own `.sky` mount, this makes sense. It seems like the best option is to use `shutil.copy2` or check if the wheel already exists before attempting to move.